phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -2376,17 +2393,42 @@ public ResultSet executeQuery(String sql) throws SQLException {
 
     @Override
     public int executeUpdate(String sql) throws SQLException {
+        CompilableStatement stmt = preExecuteUpdate(sql);
+        int updateCount = executeMutation(stmt, createAuditQueryLogger(stmt, sql));
+        flushIfNecessary();
+        return updateCount;
+    }
+
+    /**
+     * Executes the given SQL statement similar to JDBC API executeUpdate() but also returns the
+     * updated or non-updated row as Result object back to the client. This must be used with
+     * auto-commit Connection. This makes the operation atomic.
+     * If the row is successfully updated, return the updated row, otherwise if the row
+     * cannot be updated, return non-updated row.
+     *
+     * @param sql The SQL DML statement, UPSERT or DELETE for Phoenix.
+     * @return The pair of int and Result, where int represents value 1 for successful row update
+     * and 0 for non-successful row update, and Result represents the state of the row.
+     * @throws SQLException If the statement cannot be executed.
+     */
+    public Pair<Integer, Result> executeUpdateReturnRow(String sql) throws SQLException {
```

Comment:
It will be much cleaner and simpler for the application users if this API returns a ResultSet object. This is an external facing API and it is not easy to use the Result interface.

Reply:
That is correct but it requires one to one mapping of Result to ResultSet, which we don't have. I expect that we can take it up as separate improvement and change this later.

Reply:
ResultSet implementation is built entirely on iterators.

Reply:
Also, Result is single row result whereas ResultSet is not necessarily meant to be single row result because we can call `ResultSet#next` and it is meant to move the cursor forward.

Reply:
Well, it is the standard way to return information back. Single row or multiple rows doesn't matter. For example aggregate query like SELECT count(*) results are also returned through the same interface.

Reply:
Yes, i also agree with it, but what i meant is that the whole ResultSet interface in Phoenix is built on top of RowProjector and ResultIterator. We do not have support for 1:1 HBase Result to Phoenix ResultSet conversion support. It would be interesting to build one but i think it should not be scope of this work.

Reply:
At-least we could use `ResultTuple` which builds a nice wrapper over hbase Result

Reply:
Yes, used `Tuple` interface in the latest revision.